cloud/services/scalesets/client.go

```diff
@@ -207,3 +334,21 @@ func (ac *AzureClient) GetPublicIPAddress(ctx context.Context, resourceGroupName
 
 	return ac.publicIPs.Get(ctx, resourceGroupName, publicIPName, "true")
 }
+
+// NewOperationNotDoneError returns a new OperationNotDoneError wrapping a Future
+func NewOperationNotDoneError(future *infrav1.Future) *OperationNotDoneError {
```

Comment:
Should this stuff live in errors.go (in cloud pkg)? It's not scale set specific and it should be reusable by other services once we expand these changes to all reconcilers.